compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -495,83 +555,57 @@ fn subroutine_type_metadata<'ll, 'tcx>(
     )
 }
 
-// FIXME(1563): This is all a bit of a hack because 'trait pointer' is an ill-
-// defined concept. For the case of an actual trait pointer (i.e., `Box<Trait>`,
-// `&Trait`), `trait_object_type` should be the whole thing (e.g, `Box<Trait>`) and
-// `trait_type` should be the actual trait (e.g., `Trait`). Where the trait is part
-// of a DST struct, there is no `trait_object_type` and the results of this
-// function will be a little bit weird.
-fn trait_pointer_metadata<'ll, 'tcx>(
+// Create debuginfo for `dyn SomeTrait` types. Currently these are empty structs
```

Comment:
```suggestion
/// Create debuginfo for `dyn SomeTrait` types. Currently these are empty structs
```
nit: documentation comment (need to change the line below this too)